frog/imports/ui/StudentView/Runner.jsx

```diff
@@ -40,9 +42,10 @@ const Runner = ({ activity, object }) => {
 
     const RunComp = activityType.ActivityRunner;
     RunComp.displayName = activity.activityType;
-    const ActivityToRun = ReactiveHOC(activityType.dataStructure, reactiveId)(
-      RunComp
-    );
+    const ActivityToRun = ReactiveHOC(
+      cloneDeep(activityType.dataStructure),
```

Comment:
Sorry, was trying to separate out into different PRs, but this one snook in. I'll leave it in for now - this actually got removed, and the problem reappeared. Anyway, I think this is a more solid solution than either {...dataStructure} (which doesn't work for arrays), or Object.freeze, etc.